hazelcast/src/test/java/com/hazelcast/internal/management/TimedMemberStateTest.java

```diff
@@ -103,9 +109,60 @@ public void testReplicatedMapGetStats() {
     @Test
     public void testCacheGetStats() {
         NodeEngineImpl nodeEngine = getNodeEngineImpl(hz);
-        hz.getCacheManager().getCache("test-cache");
+        hz.getCacheManager().getCache(CACHE_WITH_STATS_PREFIX + "1");
         CacheService cacheService = nodeEngine.getService(CacheService.SERVICE_NAME);
         assertNotNull(cacheService.getStats()
-                          .get(CacheUtil.getDistributedObjectName("test-cache")));
+                          .get(CacheUtil.getDistributedObjectName(CACHE_WITH_STATS_PREFIX + "1")));
+    }
+
+    @Test
+    public void testAllCachesWithStatsEnabled_arePresent() {
+        NodeEngineImpl nodeEngine = getNodeEngineImpl(hz);
+        // create 100 caches with stats enabled
+        for (int i = 0; i < 100; i++) {
+            hz.getCacheManager().getCache(CACHE_WITH_STATS_PREFIX + i);
+        }
+        // create 50 caches with stats disabled
+        for (int i = 0; i < 50; i++) {
+            hz.getCacheManager().getCache(CACHE_WITHOUT_STATS_PREFIX + i);
+        }
+
+        CacheService cacheService = nodeEngine.getService(CacheService.SERVICE_NAME);
+        Map<String, LocalCacheStats> stats = cacheService.getStats();
+        for (String cacheNameWithPrefix : stats.keySet()) {
+            assertContains(cacheNameWithPrefix, CACHE_WITH_STATS_PREFIX);
+        }
```

Comment:
Don't we need an `assertEquals(100,stats.size())` too? It's verified by the next case though that the stat disabled caches actually don't have stats. Do we need then the 50 stats disabled caches in this test?

Reply:
I pushed a commit removing the redundant test.